### PR TITLE
[SHELL32] SHFileOperation: Confirm overwrite on moving files

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Confirm overwriting on moving files.
JIRA issue: [CORE-18910](https://jira.reactos.org/browse/CORE-18910)

## Proposed changes

- Add `move_file_to_file` helper function.
- Modify `move_to_dir` and `move_files` helper functions.

## TODO

- [x] Do small tests.
- [x] Do big tests.

## Screenshot

AFTER:
confirm

### CODE REVIEW

**Comment by katahiromz** [comment]
KVM BEFORE: a3e6ae2 https://reactos.org/testman/compare.php?ids=89156
KVM AFTER: 5e369c6

VBox BEFORE: a3e6ae2 https://reactos.org/testman/compare.php?ids=89157
VBox AFTER: 5e369c6

**Comment by HBelusca** [comment]
Wasn't there already a PR that was dealing with similar stuff?

File: dll/win32/shell32/shlfileop.cpp [hunk 1]
```diff
@@ -1752,7 +1776,15 @@ static void move_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, const FILE
     PathCombineW(szDestPath, feTo->szFullPath, feFrom->szFilename);
 
     if (IsAttribFile(feFrom->attributes))
+#ifdef __REACTOS__
+    {
+        WCHAR szDestPath[MAX_PATH];
+        PathCombineW(szDestPath, feTo->szFullPath, feFrom->szFilename);
```
Comment (HBelusca) [inline dll/win32/shell32/shlfileop.cpp:1782]: Why isn't it possible to just keep what's inside szDestPath? Three lines above you have already a PathCombineW call that exactly does the very same thing (and it doesn't seem that feTo/feFrom strings have changed in-between).

Comment (katahiromz) [reply dll/win32/shell32/shlfileop.cpp]: Fixed.

**Comment by katahiromz** [comment]
> Wasn't there already a PR that was dealing with similar stuff?

https://github.com/reactos/reactos/pull/5462

File: dll/win32/shell32/shlfileop.cpp [hunk 2]
```diff
@@ -1744,15 +1744,46 @@ static void move_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
...
+}
+#endif
+
 /* moves a file or directory to another directory */
 static void move_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, const FILE_ENTRY *feTo)
 {
     WCHAR szDestPath[MAX_PATH];
 
+    if (!PathFileExistsW(feTo->szFullPath))
+        SHNotifyCreateDirectoryW(feTo->szFullPath, NULL);
```
Comment (HBelusca) [inline dll/win32/shell32/shlfileop.cpp:1777]: Shouldn't that part be inside a `__REACTOS__` ifdef block?

File: dll/win32/shell32/shlfileop.cpp [hunk 3]
```diff
@@ -1744,15 +1744,46 @@ static void move_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
...
+    return SHNotifyMoveFileW(op, szFrom, szTo, FALSE) == 0;
+}
+#endif
+
 /* moves a file or directory to another directory */
 static void move_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, const FILE_ENTRY *feTo)
 {
     WCHAR szDestPath[MAX_PATH];
 
+    if (!PathFileExistsW(feTo->szFullPath))
```
Comment (katahiromz) [inline dll/win32/shell32/shlfileop.cpp:1776]:
```suggestion
#ifdef __REACTOS__
    if (!PathFileExistsW(feTo->szFullPath))
```

File: dll/win32/shell32/shlfileop.cpp [hunk 4]
```diff
@@ -1744,15 +1744,46 @@ static void move_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
...
+        SHNotifyCreateDirectoryW(feTo->szFullPath, NULL);
+
     PathCombineW(szDestPath, feTo->szFullPath, feFrom->szFilename);
```
Comment (katahiromz) [inline dll/win32/shell32/shlfileop.cpp:1779]:
```suggestion
#endif
    PathCombineW(szDestPath, feTo->szFullPath, feFrom->szFilename);
```

File: dll/win32/shell32/shlfileop.cpp [hunk 5]
```diff
@@ -1744,15 +1744,48 @@ static void move_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
...
 /* moves a file or directory to another directory */
 static void move_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, const FILE_ENTRY *feTo)
 {
     WCHAR szDestPath[MAX_PATH];
 
+#ifdef __REACTOS__
+    if (!PathFileExistsW(feTo->szFullPath))
+        SHNotifyCreateDirectoryW(feTo->szFullPath, NULL);
+
+#endif
```
Comment (katahiromz) [inline dll/win32/shell32/shlfileop.cpp:1780]:
```suggestion
#endif

```

File: dll/win32/shell32/shlfileop.cpp (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/shlfileop.cpp:1778]: Fixed.

**Pull request merged**